app/controllers/ad_steps_controller.rb

```diff
@@ -0,0 +1,50 @@
+class AdStepsController < ApplicationController
+  include Wicked::Wizard
+  steps :details, :picture, :contact
+  before_action :find_ad, only: [:show, :update]
+
+  def show
+    render_wizard
+  end  
+  
+  def update
+    case step
+    when :picture
+      @ad.update(detail_params)
+    when :contact
+      @ad.update(image_params)
+    end
+    if @ad.update(contact_params)
+      render "finish_wizard_path", notice: "Thanks for posting an AD."
+    else
+      render_wizard
+    end
+  end
+
+  def finish_wizard_path
+    ads_path
+  end
+
+
+  private
+  
+  def find_ad
+    @ad = Ad.find_by(id: params[:ad].present? ? params[:ad][:ad_id] : params[:ad_id])
+    @ad = Ad.new if @ad.blank?
```

Comment:
Can you please try the below logic? And move this ``params[:ad].present? ? params[:ad][:ad_id] : params[:ad_id]`` into `ad_id` method?

`if ad_id.present?`
    `@ad = Ad.find_by(id: ad_id)`
 `else`
   `@ad = Ad.new`
  `end`